src/witness_generator/machines/fixed_lookup_machine.rs

```diff
@@ -105,28 +105,50 @@ impl FixedLookup {
                         }
                     })
                     .collect::<Option<Vec<_>>>()
-            })
-            // deduplicate values
-            .collect::<HashSet<_>>();
+            });
 
-        let right_values = match matches.len() {
-            // no match, we error out
-            0 => {
-                return Err(EvalError::Generic("Plookup is not satisfied".to_string()));
-            }
-            // a single match, we continue
-            1 => matches.drain().next().unwrap(),
-            // multiple matches, we stop and learnt nothing
-            _ => return Ok(vec![]),
-        };
+        // fold the set of values for the individual columns into precise values and masks
+        let right_values = matches
+            .fold(None, |acc, new_value| {
```

Comment:
I reverted to using fold in this PR because for bit constraints the final mask is a fold of all matching rows. For precise values we could collect to a hashset and learn only if it had a single entry but not anymore..